libqtile/xcbq.py

```diff
@@ -316,20 +317,20 @@ def alloc_color(self, color):
         """
             Flexible color allocation.
         """
-        if color.startswith("#"):
-            if len(color) != 7:
+        try:
+            return self.conn.conn.core.AllocNamedColor(
+                self.cid, len(color), color
+            ).reply()
+        except xcffib.xproto.NameError:
+            if len(color) < 6:
                 raise ValueError("Invalid color: %s" % color)
```

Comment:
The fact is that you're avoiding `IndexError` just to raise `ValueError` :) If it were me, I'd just remove the test and the `ValueError` line, and let Python raise what's to be raised as needed, but you may want to wait for tych0 for an official opinion.